libs/datasets/timeseries.py

```diff
@@ -109,10 +109,17 @@ def source_url(self) -> Mapping[CommonFields, List[UrlStr]]:
         return source_url_series.groupby(level=0).agg(list).to_dict()
 
     def annotations(self, metric: FieldName) -> List[taglib.AnnotationWithDate]:
-        return_value = []
-        for _, row in self.tag.loc[[metric], ANNOTATION_TAG_TYPES].reset_index().iterrows():
-            return_value.append(taglib.AnnotationWithDate.make(row.tag_type, content=row.content))
-        return return_value
+        return self.tag_objects_df.loc[[metric], ANNOTATION_TAG_TYPES].to_list()
+
+    @cached_property
+    def tag_objects_df(self) -> pd.Series:
```

Comment:
is this being returned as a dataframe or a series? Type seems inconsistent with the name